ibm/resource_ibm_cis_dns_record.go

```diff
@@ -0,0 +1,540 @@
+							Computed:         true,
+							Optional:         true,
+							DiffSuppressFunc: suppressDataDiff,
+						},
+						"precision_vert": {
+							Type:             schema.TypeFloat,
+							Computed:         true,
+							Optional:         true,
+							DiffSuppressFunc: suppressDataDiff,
+						},
+						"long_direction": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"long_minutes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"long_seconds": {
+							Type:             schema.TypeFloat,
+							Computed:         true,
+							Optional:         true,
+							DiffSuppressFunc: suppressDataDiff,
+						},
+						"lat_direction": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"lat_minutes": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"lat_seconds": {
+							Type:             schema.TypeFloat,
+							Computed:         true,
+							Optional:         true,
+							DiffSuppressFunc: suppressDataDiff,
+						},
+
+						// DNSKEY record properties
+						"protocol": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"public_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						// DS record properties
+						"digest_type": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"digest": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						// NAPTR record properties
+						"order": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"preference": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"regex": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"replacement": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						// SSHFP record properties
+						"fingerprint": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						// URI record properties
+						"content": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"priority": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				DiffSuppressFunc: suppressPriority,
+			},
+
+			"proxied": {
+				Default:  false,
+				Optional: true,
+				Type:     schema.TypeBool,
+			},
+
+			"created_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"modified_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"proxiable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceIBMCISDnsRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisAPI()
+	cisId := d.Get("cis_id").(string)
+	zoneId := d.Get("domain_id").(string)
+
+	newRecord := v1.DnsBody{
+		DnsType: d.Get("type").(string),
+		Name:    d.Get("name").(string),
```

Comment:
name is an optional field so if name is not defiled it gets `nil` so is it acceptable to pass nil as Name

Reply:
Mandatory field. Changed to required: true